doc/classes/Object.xml

```diff
@@ -153,18 +181,74 @@
 				A [code]signal[/code] can only be connected once to a [code]method[/code]. It will throw an error if already connected, unless the signal was connected with [constant CONNECT_REFERENCE_COUNTED]. To avoid this, first, use [method is_connected] to check for existing connections.
 				If the [code]target[/code] is destroyed in the game's lifecycle, the connection will be lost.
 				Examples:
-				[codeblock]
-				connect("pressed", self, "_on_Button_pressed") # BaseButton signal
-				connect("text_entered", self, "_on_LineEdit_text_entered") # LineEdit signal
-				connect("hit", self, "_on_Player_hit", [ weapon_type, damage ]) # User-defined signal
-				[/codeblock]
+				[codeblocks]
+				[gdscript]
+				var button = Button.new()
+				button.pressed.connect(_on_Button_pressed) # is the same as:
+				button.connect("pressed", self, "_on_Button_pressed") # BaseButton signal
+
+				var line_edit = LineEdit.new()
+				line_edit.text_entered.connect(_on_LineEdit_text_entered) # is the same as:
+				line_edit.connect("text_entered", self, "_on_LineEdit_text_entered") # LineEdit signal
+
+				# Assuming you have a Player class that defines the `hit` signal.
+				var player = Player.new()
+				var weapon_type = "sword"
+				var damage = 100
+				player.hit.connect(_on_Player_hit.bind(weapon_type, damage)) # is the same as:
+				player.connect("hit", self, "_on_Player_hit", [weapon_type, damage])
+				[/gdscript]
+				[csharp]
+				// I am not sure if the API is bound that way. This is likely wrong
+				var button = new Button();
+				button.Pressed.Connect(new Callable(this, nameof(OnButtonPressed)); // is the same as:
+				button.Connect("pressed", this, nameof(OnButtonPressed)); // BaseButton signal
+
+				var lineEdit = new LineEdit();
+				lineEdit.TextEntered.Connect(new Callable(this, nameof(OnLineEditTextEntered)); // is the same as:
+				lineEdit.Connect("text_entered", this, nameof(OnLineEditTextEntered)); // LineEdit signal
+
+				var player = new Player(); // assuming you have a Player class that defines the hit signal.
+				string weaponType = "sword";
+				int damage = 100;
+				player.Hit.Connect(new Callable(this, nameof(OnPlayerHit)).Bind(weaponType, damage)); // is the same as:
```

Comment:
Signals as events don't support binding arguments. This is only possible the `player.Connect("hit", ...` way.